google/cloud/aiplatform/metadata/metadata.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+
+# Copyright 2021 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import logging
+from typing import Dict
+
+from google.cloud.aiplatform.metadata.metadata_store import _MetadataStore
+from google.cloud.aiplatform.metadata.context import _Context
+from google.cloud.aiplatform.metadata.execution import _Execution
+from google.cloud.aiplatform.metadata.artifact import _Artifact
+
+
+class _MetadataService:
+    """Contains the exposed APIs to interact with the Managed Metadata Service."""
+
+    def __init__(self):
+        self._experiment = None
+        self._run = None
+
+    def set_experiment(self, experiment: str):
+        if not experiment:
+            raise ValueError(f"Invalid experiment {experiment}.")
+
+        store = _MetadataStore.get()
+        if not store:
+            logging.info(
+                f"Creating a default MetadataStore for experiment {experiment}"
+            )
+            _MetadataStore.create()
+
+        context = _Context.get(resource_name=experiment)
+        if not context:
+            logging.info(f"Creating a Context for experiment {experiment}")
+            context = _Context.create(
+                resource_id=experiment,
+                schema_title="system.Experiment",
+                schema_version="0.0.1",
+            )
+        self._experiment = context.name
+
+    def set_run(self, run: str):
+        if not self._experiment:
+            raise ValueError(
+                "No experiment set for this run. Make sure to call aiplatform.init(experiment='my-experiment') or "
+                "aiplatform.set_experiment(experiment='my-experiment') before trying to set_run. "
+            )
+        if not run:
+            raise ValueError(f"Invalid run {run}.")
+
+        execution = _Execution.get(resource_name=run)
+        if not execution:
+            logging.info(f"Creating an Execution for run {run}")
+            execution = _Execution.create(
+                resource_id=run, schema_title="system.Run", schema_version="0.0.1",
+            )
+        self._run = execution.name
+
+    def log_param(self, name: str, value):
+        if not isinstance(value, (float, int, str)):
+            raise ValueError(
+                f"Invalid type for parameter value {value}. It should be one of [int, float, str]"
+            )
+        return self.log_params({name: value})
+
+    def log_params(self, params: Dict):
+        if not self._experiment:
```

Comment:
Capturing that the preference is to automatically create experiment and run if one has not been set.

Reply:
@jialuzh mentioned the same. He is going to clean this up a bit. Seems that Nate doesn't want the set_run, nor other stuff. If you don't mind, we can leave it as is for now, and further discuss?
I'm more of the idea of actually not automagically doing things. IMO the user should have done an init or set_experiment, set_run before... But I have no strong opinion..

Reply:
Let's actually leave this for a follow up once we agree with Nate tomorrow... SG?

Reply:
SGTM